wallet/wallet.go

```diff
@@ -301,6 +302,34 @@ func (w *SingleAddressWallet) Redistribute(cs consensus.State, outputs int, amou
 	w.mu.Lock()
 	defer w.mu.Unlock()
 
+	// build map of inputs currently in the tx pool
+	inPool := make(map[types.Hash256]bool)
+	for _, ptxn := range pool {
+		for _, in := range ptxn.SiacoinInputs {
+			inPool[types.Hash256(in.ParentID)] = true
+		}
+	}
+
+	// fetch unspent transaction outputs
+	utxos, err := w.store.UnspentSiacoinElements(false)
+	if err != nil {
+		return types.Transaction{}, nil, err
+	}
+
+	// check whether a redistribution is necessary, adjust number of desired
+	// outputs accordingly
+	for _, sce := range utxos {
+		inUse := w.isOutputUsed(sce.ID) || inPool[sce.ID]
+		matured := cs.Index.Height >= sce.MaturityHeight
+		sameValue := sce.Value.Equals(amount)
+		if !inUse && matured && sameValue {
+			outputs--
+		}
+	}
+	if outputs <= 0 {
+		return types.Transaction{}, nil, api.ErrWalletAlreadyRedistributed
```

Comment:
Instead of a custom error I'd just return `nil`. If we change the definition of `Redistribute` to "making sure that at least 'outputs' outputs exist with >= 'amount' value each" then it probably makes sense to not consider this an error.

Reply:
Yeah true. I'll change it. I did it because I wanted to log it in the autopilot so we're aware when this happens but I guess that's not really useful info anyways.